numpy/random/_generator.pyx

```diff
@@ -4416,6 +4416,8 @@ cdef class Generator:
         else:
             if type(out) is not np.ndarray:
                 raise TypeError('out must be a numpy array')
+            if not (np.PyArray_FLAGS(out) & np.NPY_ARRAY_WRITEABLE):
```

Comment:
This should use `PyArray_FailUnlessWriteable`, I think.  Because that is a bit safer/better in rarer cases when we transition from readonly to writeonly.

Reply:
Thanks, @seberg, I force-pushed an update that uses `PyArray_FailUnless Writeable`.